### PR TITLE
Actions update

### PR DESCRIPTION
**Pull request by jgustie**
These actions were pretty old, this cleans them up and makes it easier to add new images.

### CODE REVIEW

**Comment by jgustie** [comment]
Note that this changes the action names and structure and will require an update to the branch protection rules; I can fix that after merging.

**Comment by tisba** [comment]
There might be some more changes incoming via https://github.com/thestormforge/optimize-trials/pull/7. Will have to take a look /cc @ztreinhart

**Comment by jgustie** [comment]
I would expect that you would rebase #7 on top of this once merged and just add the new image (which will be a lot easier now).

**Comment by tisba** [comment]
> I would expect that you would rebase #7 on top of this once merged and just add the new image (which will be a lot easier now).

yeah sure. Since I'm in a bunch of meetings now, I just wanted to bring that on Zach's radar :)

File: .github/workflows/build-push.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    - name: Build and push
+      id: docker_build
+      uses: docker/build-push-action@v3
+      with:
+        context: ./${{ inputs.dir }}
+        file: ./${{ inputs.dir }}/Dockerfile
+        push: true
+        tags: ${{ steps.docker_meta.outputs.tags }}
+        labels: ${{ steps.docker_meta.outputs.labels }}
```
Comment (tisba) [inline .github/workflows/build-push.yaml:57]: Do we want to add caching? (see e.g. https://github.com/thestormforge/optimize-trials/pull/7/files#diff-ee31744c16c09322dc783a319da3c759dd38bf18f444b8f5f1ed0fb4ef49e65cR33-R34).

**Review by ztreinhart — APPROVED** [review]
This seems like a good improvement. Thanks @jgustie. 👍

**Pull request merged**